test/jtreg-ext/requires/VMProps.java

```diff
@@ -135,6 +140,7 @@ public Map<String, String> call() {
         map.put("jdk.containerized", this::jdkContainerized);
         map.put("vm.flagless", this::isFlagless);
         map.put("jdk.foreign.linker", this::jdkForeignLinker);
+        map.putAll(xOptFlags()); // -Xmx4g -> @requires vm.opt.x.Xmx == "4g" )
```

Comment:
I know of no colliding options ATM. They could be colliding in the future (especially if we chose to parse extra options not starting with a `-X`). Also, I think it is incredibly confusing to reuse the `vm.opt` namespace that is used by vanilla JTREG by our extension code in VMProps. I would object to put both in the `vm.opt` namespace.

Reply:
I agree it is a bit ugly and the point of this being an extension is good. An alternative would be to use `vm.xopt.Xmx` to even less clobber the `vm.opt` namespace, but I guess we already do by adding the `vm.opt.final` details. 

I'm ok with the current way, just wanted to raise possible alternatives.

Reply:
`vm.xopt` was my first choice, but it works badly because of other reasons, so lets keep it as it is now.

Reply:
The reason is that we need to use a namespace that starts with `vm.opt`, because that string is *special* in JTREG. If the property starts with that prefix, there is no need to register all possible values manually --- something that would be hard to maintain. However, reusing parts of namespaces have prior art in for example `vm.gc.*` so although maybe not optimal, it should not be problematic. It follows standard practice.